lesson_9/homework_09.py

```diff
@@ -128,4 +128,7 @@ def test_empty_substring(self):
         self.assertEqual(result, expected)
 
 if __name__ == '__name__':
```

Comment:
For your file it is no necessary to create several in __name__ == '__main__'. It is still enough for 1 global initialization for whole file. And you make a mistake in your last expression of name == name instead main)